src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentitySource.cs

```diff
@@ -29,14 +30,31 @@ protected ManagedIdentitySource(RequestContext requestContext)
 
         public virtual async Task<ManagedIdentityResponse> AuthenticateAsync(AppTokenProviderParameters parameters, CancellationToken cancellationToken)
         {
-            ManagedIdentityRequest request = CreateRequest(parameters.Scopes.ToArray());
+            // Convert the scopes to a resource string.
+            string resource = ScopeHelper.ScopesToResource(parameters.Scopes.ToArray());
 
-            HttpResponse response =
+            if (cancellationToken.IsCancellationRequested)
+            {
+                _requestContext.Logger.Error(TimeoutError);
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+
+            ManagedIdentityRequest request = CreateRequest(resource);
+
+            try
+            {
+                HttpResponse response =
             request.Method == HttpMethod.Get ?
-            await _requestContext.ServiceBundle.HttpManager.SendGetForceResponseAsync(request.Endpoint, request.Headers, _requestContext.Logger, cancellationToken: cancellationToken).ConfigureAwait(false) :
-            await _requestContext.ServiceBundle.HttpManager.SendPostForceResponseAsync(request.Endpoint, request.Headers, request.BodyParameters, _requestContext.Logger, cancellationToken: cancellationToken).ConfigureAwait(false);
+            await _requestContext.ServiceBundle.HttpManager.SendGetForceResponseAsync(request.ComputeUri(), request.Headers, _requestContext.Logger, cancellationToken: cancellationToken).ConfigureAwait(false) :
```

Comment:
Is it possible to compute the uri only once and store it as a property rather than computing it multiple times.